src/api.rs

```diff
@@ -1,33 +1,151 @@
 // API endpoints
 
-use actix_web::{post, get, Responder, web::Payload};
+use std::fmt::{Display, Formatter, write};
+use std::ops::DerefMut;
+use std::str::FromStr;
+use std::sync::RwLock;
+use actix_web::{post, get, Responder, web, Result, ResponseError, Error};
+use actix_web::http::StatusCode;
+use actix_web::web::Json;
+use serde::{Serialize, Deserialize};
+use serde_json::{json};
+use crate::api::ValidationError::NoSuchGameError;
+use crate::game::{GameManager, adapter, connect4, GameId};
+use crate::game::adapter::GameAdapter;
+
+/* Helpers */
+
+/// ValidationError
+#[derive(Debug, Clone)]
+pub enum ValidationError {
+    ParseIntError(String),
+    NoSuchGameError(String)
+}
+
+impl Display for ValidationError {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        match self {
+            ValidationError::ParseIntError(id) => write!(f, "ID {} is Invalid", id),
```

Comment:
Should this be called a ParseIDError in case we change the format of IDs?

Reply:
Ok